### PR TITLE
Use mitochondria::OnceCell to store ScriptThread in TLS

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15136)

### CODE REVIEW

**Comment by nox** [comment]
```
Tests with unexpected results:
  ▶ TIMEOUT [expected ERROR] /navigation-timing/nav2_test_unloadEvents_previous_document_cross_origin.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1
```

**Comment by nox** [comment]
Checked locally, this seems to be a new intermittent. Should I just file it as such, @jdm?

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -534,29 +535,26 @@ impl ScriptThreadFactory for ScriptThread {
...
-            let mut failsafe = ScriptMemoryFailsafe::new(&script_thread);
+                let mut failsafe = ScriptMemoryFailsafe::new(&script_thread);
 
-            let origin = Origin::new(&load_data.url);
-            let new_load = InProgressLoad::new(id, frame_id, parent_info, layout_chan, window_size,
-                                               load_data.url.clone(), origin);
-            script_thread.start_page_load(new_load, load_data);
+                let origin = Origin::new(&load_data.url);
+                let new_load = InProgressLoad::new(id, frame_id, parent_info, layout_chan, window_size,
+                                                load_data.url.clone(), origin);
```
Comment (Ms2ger) [inline components/script/script_thread.rs]: Indentation

**Comment by nox** [comment]
@Ms2ger Amended.
@jdm Filed #15151.

@bors-servo r=jdm

**Comment by mbrubeck** [comment]
Reverted in #15176 in an attempt to fix #15161.

**Pull request merged**